woocommerce/woocommerce-ios
# Lightweight storefront: Install theme from Theme preview screen

Closes: #11494 


## Description

Performs theme installation from the theme preview screen if accessed from theme settings screen. 

Changes
- Inject siteID and theme to view model to perform theme installation.
- Perform theme installation if mode is theme settings. 
- Handling loading and error state for theme installation. 
- Rename "Start with This Theme" button into "Use this theme" for theme settings screen.

## Testing instructions

Profiler flow foreground
- Launch the app
- Navigate to store picker
- Create a new store and select a theme of your choice
- Tap on the "Start with This Theme" button
- Wait for the store creation to finish
- Once you land on the dashboard, the theme installation should happen silently in the background
- Navigate to Settings -> Themes and validate that your theme is installed

Profiler flow background
- Launch the app
- Navigate to store picker
- Create a new store and select a theme of your choice
- You should be navigated to store creation loading screen (with Woo-related facts)
- Kill the app
- Relaunch the app
- Wait for a few inside the app
- You will see an alert that the new store is ready
- Switch to the new store from the alert
- The theme installation should happen silently in the background
- Navigate to Settings -> Themes and validate that your theme is installed

Theme installation
- Launch the app
- Navigate to Settings -> Themes 
- Select a theme
- From the theme preview screen, tap on the "Use this theme" button
- Theme should be installed 
- Select a different theme 
- Once you are in the theme preview screen turn off internet
- Tap on "Use this theme" button
- Validate that an error notice is displayed when theme installation fails

## Screenshots

#### Profiler flow

https://github.com/woocommerce/woocommerce-ios/assets/524475/8875ce29-d0ad-4e5f-b2b8-1dbc8836fa77

#### Theme settings flow

https://github.com/woocommerce/woocommerce-ios/assets/524475/6c9715df-cf71-4abe-9d0e-81da0e44b37c


---
